kiva/qpainter.py

```diff
@@ -553,48 +553,27 @@ def radial_gradient(self, cx, cy, r, fx, fy, stops, spread_method,
 
     def draw_image(self, img, rect=None):
         """
-        img is either a N*M*3 or N*M*4 numpy array, or a Kiva image
+        img is either a N*M*3 or N*M*4 numpy array, or a PIL Image
 
         rect - a tuple (x, y, w, h)
         """
-        from kiva import agg
-
-        def copy_padded(array):
-            """ Pad image width to a multiple of 4 pixels, and minimum dims of
-                12x12. QImage is very particular about its data.
-            """
-            y, x, d = array.shape
-            pad = (lambda v: (4 - (v % 4)) % 4)
-            nx = max(x + pad(x), 12)
-            ny = max(y, 12)
-            if x == nx and y == ny:
-                return array
-            ret = np.zeros((ny, nx, d), dtype=np.uint8)
-            ret[:y, :x] = array[:]
-            return ret
```

Comment:
Just to confirm, we don't need to copy and pad the array data - because the use of `PIL.Image.Image` means that this is implicitly handled. Correct?

Reply:
We do, but PIL handles it for us now.